src/silicone/multiple_infillers/decompose_collection_with_time_dep_ratio.py

```diff
@@ -73,6 +73,19 @@ def _construct_consistent_values(self, aggregate_name, components, db_to_generat
         use["variable"] = aggregate_name
         return pyam.IamDataFrame(use)
 
+    def _set_of_units_without_equiv(self, df):
+        """
+        Parameters
+        ----------
+        df : obj:`pyam.IamDataFrame`
+            The dataframe whose units we want
+
+        Returns
+        -------
+        Set
+            The set of units
+        """
```

Comment:
Could you make it more explicit what kind of set of units is returned?

Reply:
I'll write more for this anyway